xhr.bs

```diff
@@ -1018,6 +1024,8 @@ return <a>this</a>'s <a>cross-origin credentials</a>.
 <ol>
  <li><p>Set <var>xhr</var>'s <a>state</a> to <i>done</i>.
 
+ <li><p><a for=/>Report timing</a> for <var>xhr</var>.
```

Comment:
We report timings here, but two steps down we set response to a network error. Whereas above we don't report timing at all if response is a network error.

Now per https://github.com/whatwg/fetch/pull/1202 maybe some network errors should get report timing, but I don't think the current setup makes that happen correctly, right?

Reply:
We always call "Report timing", and let FETCH decide if to report or not (if the response has a `timing info`. WDYM by "above we don't report timing at all if response is a network error", I think we do: ` <var>xhr</var>'s <a for=XMLHttpRequest>response</a>'s is a
 <a for=/>network error</a>, run the <a>request error steps</a> for <var>xhr</var>` (request error steps would report the timing).

Reply:
Yeah, I think that's right. Let's solve https://github.com/whatwg/fetch/issues/1215 before landing this then as currently finalize and report timing always returns early for network errors I just discovered.

I think it would also be good to ensure we have test coverage here. In the sense that the final readystatechange/load event has access to the RT entry.

Reply:
> Let's solve [whatwg/fetch#1215](https://github.com/whatwg/fetch/issues/1215) before landing this then as currently finalize and report timing always returns early for network errors I just discovered.
> 

https://github.com/whatwg/fetch/pull/1311 addressed this.

Reply:
Added a requisite test here: https://github.com/web-platform-tests/wpt/pull/31027